L1 Canned Metrics for Kinesis don't compile

```
aws-kinesis/lib/kinesis-fixed-canned-metrics.ts:60:17 - error TS4050: Return type of public static method from exported class has or is using name 'MetricWithDims' from external module "/home/runner/work/awscdk-service-spec/awscdk-service-spec/aws-cdk/packages/aws-cdk-lib/aws-kinesis/lib/kinesis-canned-metrics.generated" but cannot be named.

60   public static getRecordsIteratorAgeMillisecondsMaximum(dimensions: { StreamName: string }) {
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
aws-kinesis/lib/stream.ts:429:42 - error TS2345: Argument of type '(dimensions: { StreamName: string; }) => MetricWithDims<{ StreamName: string; }>' is not assignable to parameter of type '(dimensions: { StreamName: string; }) => MetricProps'.
  Call signature return types 'MetricWithDims<{ StreamName: string; }>' and 'MetricProps' are incompatible.
    The types of 'namespace' are incompatible between these types.
      Type 'string | undefined' is not assignable to type 'string'.
        Type 'undefined' is not assignable to type 'string'.

429     return this.metricFromCannedFunction(KinesisMetrics.getRecordsIteratorAgeMillisecondsMaximum, props);
                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
aws-kinesis/lib/stream.ts:499:42 - error TS2345: Argument of type '(dimensions: { StreamName: string; }) => { statistic: string; namespace?: string | undefined; metricName: string; dimensionsMap: { StreamName: string; }; }' is not assignable to parameter of type '(dimensions: { StreamName: string; }) => MetricProps'.
  Call signature return types '{ statistic: string; namespace?: string | undefined; metricName: string; dimensionsMap: { StreamName: string; }; }' and 'MetricProps' are incompatible.
    The types of 'namespace' are incompatible between these types.
      Type 'string | undefined' is not assignable to type 'string'.
        Type 'undefined' is not assignable to type 'string'.

499     return this.metricFromCannedFunction(KinesisMetrics.putRecordSuccessAverage, props);
                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
aws-kinesis/lib/stream.ts:510:42 - error TS2345: Argument of type '(dimensions: { StreamName: string; }) => { statistic: string; namespace?: string | undefined; metricName: string; dimensionsMap: { StreamName: string; }; }' is not assignable to parameter of type '(dimensions: { StreamName: string; }) => MetricProps'.
  Call signature return types '{ statistic: string; namespace?: string | undefined; metricName: string; dimensionsMap: { StreamName: string; }; }' and 'MetricProps' are incompatible.
    The types of 'namespace' are incompatible between these types.
      Type 'string | undefined' is not assignable to type 'string'.
        Type 'undefined' is not assignable to type 'string'.

510     return this.metricFromCannedFunction(KinesisMetrics.putRecordsBytesAverage, props);
                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
aws-kinesis/lib/stream.ts:594:42 - error TS2345: Argument of type '(dimensions: { StreamName: string; }) => { statistic: string; namespace?: string | undefined; metricName: string; dimensionsMap: { StreamName: string; }; }' is not assignable to parameter of type '(dimensions: { StreamName: string; }) => MetricProps'.
  Call signature return types '{ statistic: string; namespace?: string | undefined; metricName: string; dimensionsMap: { StreamName: string; }; }' and 'MetricProps' are incompatible.
    The types of 'namespace' are incompatible between these types.
      Type 'string | undefined' is not assignable to type 'string'.
        Type 'undefined' is not assignable to type 'string'.

594     return this.metricFromCannedFunction(KinesisMetrics.incomingBytesAverage, props);
                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
aws-kinesis/lib/stream.ts:607:42 - error TS2345: Argument of type '(dimensions: { StreamName: string; }) => { statistic: string; namespace?: string | undefined; metricName: string; dimensionsMap: { StreamName: string; }; }' is not assignable to parameter of type '(dimensions: { StreamName: string; }) => MetricProps'.
  Call signature return types '{ statistic: string; namespace?: string | undefined; metricName: string; dimensionsMap: { StreamName: string; }; }' and 'MetricProps' are incompatible.
    The types of 'namespace' are incompatible between these types.
      Type 'string | undefined' is not assignable to type 'string'.
        Type 'undefined' is not assignable to type 'string'.

607     return this.metricFromCannedFunction(KinesisMetrics.incomingRecordsAverage, props);
                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
aws-kinesis/lib/stream.ts:626:42 - error TS2345: Argument of type '(dimensions: { StreamName: string; }) => { statistic: string; namespace?: string | undefined; metricName: string; dimensionsMap: { StreamName: string; }; }' is not assignable to parameter of type '(dimensions: { StreamName: string; }) => MetricProps'.
  Call signature return types '{ statistic: string; namespace?: string | undefined; metricName: string; dimensionsMap: { StreamName: string; }; }' and 'MetricProps' are incompatible.
    The types of 'namespace' are incompatible between these types.
      Type 'string | undefined' is not assignable to type 'string'.
        Type 'undefined' is not assignable to type 'string'.

626     return this.metricFromCannedFunction(KinesisMetrics.readProvisionedThroughputExceededAverage, props);
                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
aws-kinesis/lib/stream.ts:644:42 - error TS2345: Argument of type '(dimensions: { StreamName: string; }) => { statistic: string; namespace?: string | undefined; metricName: string; dimensionsMap: { StreamName: string; }; }' is not assignable to parameter of type '(dimensions: { StreamName: string; }) => MetricProps'.
  Call signature return types '{ statistic: string; namespace?: string | undefined; metricName: string; dimensionsMap: { StreamName: string; }; }' and 'MetricProps' are incompatible.
    The types of 'namespace' are incompatible between these types.
      Type 'string | undefined' is not assignable to type 'string'.
        Type 'undefined' is not assignable to type 'string'.

644     return this.metricFromCannedFunction(KinesisMetrics.writeProvisionedThroughputExceededAverage, props);
                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/packages/@aws-cdk/cfn-resources/src/cli/cdk/canned-metrics.ts
+++ b/packages/@aws-cdk/cfn-resources/src/cli/cdk/canned-metrics.ts
@@ -59,12 +59,12 @@ export class MetricsClass extends ClassType {
 
     this.returnType = new InterfaceType(this.scope, {
       name: 'MetricWithDims',
+      export: true,
       properties: [
         {
           name: 'namespace',
           type: Type.STRING,
           immutable: true,
-          optional: true,
         },
         {
           name: 'metricName',
@@ -98,7 +98,7 @@ export class MetricsClass extends ClassType {
     }
 
     // If we have more than one dimension set, add a generic declaration
-    if (dimensionSets.length >= 1) {
+    if (dimensionSets.length > 1) {
       this.addMetricMethodDeclaration(name, Type.ANY);
     }
 

